client/my-sites/site-settings/section-import.jsx

```diff
@@ -178,23 +196,32 @@ class SiteSettingsImport extends Component {
 			api: { isHydrated },
 			importers: imports,
 		} = this.state;
-		const { site } = this.props;
-		const { slug, title } = site;
-		const siteTitle = title.length ? title : slug;
+		// TODO: site and siteTitle should be taken from props where needed,
+		// rather than passed along to methods
+		const { importerOption, site } = this.props;
+		const siteTitle = getSiteTitle( site );
 
 		if ( ! isHydrated ) {
 			return this.renderIdleImporters( site, siteTitle, appStates.DISABLED );
 		}
 
-		const importsForSite = filterImportsForSite( site.ID, imports )
-			// Add in the 'site' and 'siteTitle' properties to the import objects.
-			.map( item => Object.assign( {}, item, { site, siteTitle } ) );
+		const activeImporter = flow(
+			// Pick only imports for the current site
+			items => filterImportsForSite( site.ID, items ),
+			// Add in the 'site' and 'siteTitle' properties
+			items =>
+				items.map( item => ( {
+					...item,
+					site,
+					siteTitle,
+				} ) ),
+			// Find the first enabled import
+			items => find( items, item => includes( enabledImporterTypes, item.type ) )
+		)( imports );
 
-		if ( 0 === importsForSite.length ) {
-			return this.renderIdleImporters( site, siteTitle, appStates.INACTIVE );
-		}
-
-		return this.renderActiveImporters( importsForSite );
+		return activeImporter || importerOption
+			? this.renderImporterScreen( activeImporter )
```

Comment:
Did you intend to call `this.renderImporterScreen( importerOption )` when `activeImporter` is not truthy but `importerOption` is?

Reply:
Unless I'm missing something, `importerOption` never gets passed into `renderImporterScreen`.  Is it derived out of other state or something?

Reply:
Ahh you're right. It was using the `importerOption` before, but I tweaked it recently so that it'd meet the needs of this `activeImporter`. Now there's always an `activeImporter` as before.
I'm not really sure what the direction is anymore since it keeps changing but if we still want to get rid of the need for this fake importer instance then this PR will need a bit more work I think.

Reply:
Actually, sorry - the `importerOption` is used [here](https://github.com/Automattic/wp-calypso/pull/26723/files#diff-3d171fe186cbf9076a950ccb28901647R164) when `activeImporter` is missing.